apps/sports-helsinki/src/domain/search/combinedSearch/hooks.ts

```diff
@@ -0,0 +1,163 @@
+      variables,
+      // FIXME: Set the fetch policy to not trigger cache,
+      // or the Apollo client fails to request
+      // same query multiple times. Depending on settings,
+      // the client stops the query when the first response is given,
+      // and uses the cache for the next query with same name or
+      // returns the items twice.
+      // Since the query is exactly the same as the actual listing query,
+      // it would be really nice to have the result in cache already.
+      // See https://community.apollographql.com/t/executing-the-same-query-multiple-times-with-different-variables/3052
+      fetchPolicy: 'no-cache',
+      ssr: false,
+    }
+  );
+
+  React.useEffect(() => {
+    const count = () => {
+      if (loading || !data) {
+        return null;
+      }
+      return (data as SearchListQuery)?.unifiedSearch?.count ?? 0;
+    };
+    setResultCount('Venue', count());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data, loading]);
+
+  return { search, loading, data, ...delegatedProps };
+}
+
+export function useLazyEventSearchForTabCount({
+  eventType,
+}: {
+  eventType: EventTypeId;
+}) {
+  const { setResultCount } = useTabsContext();
+  const variables = useEventSearchFilters(eventType);
+  const [search, { loading, data, ...delegatedProps }] = useLazyQuery(
+    EventListDocument,
+    {
+      variables,
+      // FIXME: Set the fetch policy to not trigger cache,
+      // or the Apollo client fails to request
+      // same query multiple times. Depending on settings,
+      // the client stops the query when the first response is given,
+      // and uses the cache for the next query with same name or
+      // returns the items twice.
+      // Since the query is exactly the same as the actual listing query,
+      // it would be really nice to have the result in cache already.
+      // See https://community.apollographql.com/t/executing-the-same-query-multiple-times-with-different-variables/3052
+      fetchPolicy: 'no-cache',
+      ssr: false,
+    }
+  );
+
+  React.useEffect(() => {
+    const count = () => {
+      if (loading || !data) {
+        return null;
+      }
+      return ((data as EventListQuery)?.eventList?.meta.count as number) ?? 0;
+    };
+    setResultCount(eventType, count());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data, loading, eventType]);
+
+  return { search, loading, data, ...delegatedProps };
+}
+
+export function useSearchTabResultCounts() {
+  const { search: venueSearch } = useLazyVenueSearchForTabCount();
+  const { search: generalEventSearch } = useLazyEventSearchForTabCount({
+    eventType: EventTypeId.General,
+  });
+  const { search: courseEventSearch } = useLazyEventSearchForTabCount({
+    eventType: EventTypeId.Course,
+  });
+  React.useEffect(() => {
+    venueSearch();
+    generalEventSearch();
+    courseEventSearch();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+}
```

Comment:
Maybe all these hooks could be as separate files under `hooks/`, just so they follow same logic like elsewhere in the project?

Reply:
Yup I think all the other hooks are always located on `src/hooks` on every app and package